va/va.go

```diff
@@ -623,9 +624,14 @@ func (va *ValidationAuthorityImpl) performLocalValidation(
 	return records, nil
 }
 
-// PerformValidation validates the challenge for the domain in the request.
-// The returned result will always contain a list of validation records, even
-// when it also contains a problem.
+// PerformValidation conducts a local Domain Control Validation (DCV) and CAA
+// check for the specified challenge and dnsName. When invoked on the primary
+// Validation Authority (VA) and the local validation succeeds, it also performs
+// DCV and CAA checks using the configured remote VAs. The method returns a
+// ValidationResult and an error if the validation fails. The ValidationResult
```

Comment:
This makes it sound like it returns both a result _and_ and error if validation fails. By my reading that is not the case: it returns "nil, err" in a few cases (error validating inputs, error marshalling a PB), and returns "record (maybe with internal problem), nil" in all other cases.

Reply:
You're right, this can be improved.